### DISCUSSION TITLE
Code for IBM i and BOB integration - PATH problem

### DISCUSSION THREAD

**Discussion by beckhamrryyaann** [codefori/vscode-ibmi#1088] [General]
I am trying to integrate Code for IBM i with BOB.  I am developing locally and have my action created to "deploy and build".
When I try to run the action, I get an error saying in Code for IBM i output:
 `It looks like /QOpenSys/pkgs/bin/ is not currently in your system PATH. This means that programs and tools installed in this directory may not be easily accessible from the command line.`

My profile is using the bash shell, I have the /QOpenSys/pkgs/bin/ in my path in the .bash_profile file.   I can run the makei build command from the pase command line with no problem.    But I cannot run it inside "Code for IBM i".

![image](https://user-images.githubusercontent.com/6051612/220756949-83e31440-006d-4273-8dc8-3f7f236e3322.png)

Here is my PATH from the pase terminal inside vs code
![image](https://user-images.githubusercontent.com/6051612/220757304-ef370658-b2cb-4834-ae6c-520e9c7b42f0.png)

I am not sure what to try next?  I had trouble with this with the last project I created using BOB and code for IBM i but i just used the command line to build.  But I would really like to start using the actions inside vs code so it will be easier on the other developers who are not as familiar with these more modern ways.

Ryan

**Comment by sebjulliand** [comment]
Hi @beckhamrryyaann!
Could you try to set your PATH in a `.profile` file instead of `.bash_profile`? I think `.bash_profile` is not run when not logging into a shell, but `.profile` is in all cases.

**Reply by beckhamrryyaann** [reply]
Thank you for your reply.  That did not fix the error.  Even the error message i get says to put the path in the .bash_profile file.

**Reply by worksofliam** [reply]
@sebjulliand You were fast! That was also my go-to idea.

I think the issue that the non-interactive shell running your Action is not inheriting your `.bash_profile`. How did you configure your default shell?

Note, when you change any of those dot files, the connection inside of Code for IBM i needs to be restarted for them to be picked up.

**Reply by sebjulliand** [reply]
Hmmm...try to set the path in a `.bashrc` file then (with execution rights). On my setup I have a `.profile` and a `.bashrc` file.

**Reply by beckhamrryyaann** [reply]
@sebjulliand I had to have a .bashrc file.   Thanks for the help. If not already, that bit of information should probably be added to the docs.
